openedx/features/calendar_sync/__init__.py

```diff
@@ -3,15 +3,16 @@
 """
 
 
-def get_calendar_event_id(user, block_key, date_type):
+def get_calendar_event_id(user, block_key, date_type, hostname):
     """
     Creates a unique event id based on a user and a course block key
 
     Parameters:
         user (User): The user requesting a calendar event
         block_key (str): The block key containing the date for the calendar event
         date_type (str): The type of the date (e.g. 'due', 'start', 'end', etc.)
+        hostname (str): A hostname to namespace this id (e.g. 'open.edx.org')
     Returns:
         event id (str)
     """
-    return user.username + '.' + block_key + '.' + date_type
+    return '{}.{}.{}@{}'.format(user.id, block_key, date_type, hostname)
```

Comment:
ics spec recommends using a hostname, since these values must be unique globally, across all events on the user's calendar.

I also switched from username to user id, since I think we generally prefer to avoid usernames when possible.